examples/weather_station/src/battery.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "battery.h"
+
+#include <hal/nrf_saadc.h>
+
+#include <drivers/adc.h>
+#include <drivers/gpio.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(app);
+
+#define VBATT DT_PATH(vbatt)
+#define BATTERY_ADC_DEVICE_NAME DT_LABEL(DT_NODELABEL(adc))
+#define BATTERY_ENABLE_MEASUREMENT_GPIO DT_GPIO_LABEL(VBATT, power_gpios)
+#define BATTERY_ENABLE_MEASUREMENT_PIN DT_GPIO_PIN(VBATT, power_gpios)
+#define BATTERY_ENABLE_MEASUREMENT_FLAGS DT_GPIO_FLAGS(VBATT, power_gpios)
+#define BATTERY_CHARGE_GPIO DT_LABEL(DT_NODELABEL(gpio1))
+#define BATTERY_CHARGE_PIN 0
+#define BATTERY_FULL_OHMS DT_PROP(VBATT, full_ohms)
+#define BATTERY_OUTPUT_OHMS DT_PROP(VBATT, output_ohms)
+
+const device *kAdcController;
```

Comment:
These are technically not constants, so shouldn't have `k` prefix. Also, put them in anonymous namespace.